lib/spack/spack/package.py

```diff
@@ -1428,6 +1578,114 @@ def rpath_args(self):
         return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
 
 
+class Package(PackageBase):
+    phases = ['install']
+    # This will be used as a registration decorator in user
+    # packages, if need be
+    PackageBase.sanity_check('install')(PackageBase.sanity_check_prefix)
+
+
+class EditableMakefile(PackageBase):
```

Comment:
should not this be derived from AutotoolsPackage? Why do we actually need a separate class for this? Maybe just add this single `edit()` method to AutotoolsPackage with default doing nothing?